data/src/xet_file.rs

```diff
@@ -0,0 +1,43 @@
+use merklehash::{DataHashHexParseError, MerkleHash};
+use serde::{Deserialize, Serialize};
+use tracing::error;
+
+/// A struct that wraps a the Xet file information.
+#[derive(Debug, Clone, PartialEq, Eq, Default, Serialize, Deserialize)]
+pub struct XetFileInfo {
+    /// The Merkle hash of the file
+    hash: String,
+
+    /// The size of the file
+    file_size: usize,
+}
+
+impl XetFileInfo {
+    /// Creates a new `XetFileInfo` instance.
+    ///
+    /// # Arguments
+    ///
+    /// * `hash` - The Xet hash of the file. This is a Merkle hash string.
+    /// * `file_size` - The size of the file.
+    pub fn new(hash: String, file_size: usize) -> Self {
+        Self { hash, file_size }
+    }
+
+    /// Returns the Merkle hash of the file.
+    pub fn hash(&self) -> &str {
+        &self.hash
+    }
+
+    /// Returns the parsed merkle hash of the file.
+    pub fn merkle_hash(&self) -> std::result::Result<MerkleHash, DataHashHexParseError> {
+        MerkleHash::from_hex(&self.hash).map_err(|e| {
+            error!("Error parsing hash value for file info {e:?}");
+            e
+        })
```

Comment:
[nit] use `log_error` for this
```suggestion
        MerkleHash::from_hex(&self.hash).log_error("Error parsing hash value for file info")
```